src/main/webapp/intrigue-api/search-forms/search-forms.js

```diff
@@ -0,0 +1,141 @@
+const ROOT = '/search/catalog/internal'
+
+const typeDefs = `
+  type QueryTemplate {
+    security_access_administrators: [String]
+    security_access_groups: [String]
+    security_access_groups_read: [String]
+    security_access_individuals: [String]
+    security_access_individuals_read: [String]
+    created: Date
+    modified: Date
+    filter_template: Json
+    owner: String
+    query_settings: QuerySettings
+    id: ID!
+    title: String
+  }
+
+  input QueryTemplateInput {
+    security_access_administrators: [String]
+    security_access_groups: [String]
+    security_access_groups_read: [String]
+    security_access_individuals: [String]
+    security_access_individuals_read: [String]
+    created: Date
+    modified: Date
+    filter_template: Json
+    owner: String
+    query_settings: QuerySettingsInput
+    id: ID!
+    title: String
+  }
+
+  extend type Query {
+    queryTemplates: [QueryTemplate]
+  }
+
+  extend type Mutation {
+    deleteQueryTemplate(id: ID!): ID
+    saveQueryTemplate(queryTemplate: QueryTemplateInput): QueryTemplate
+  }
+`
+
+const inverseKeys = object => {
+  return Object.keys(object).reduce((ret, key) => {
+    ret[object[key]] = key
+    return ret
+  }, {})
+}
+
+const transformAttrs = {
+  accessAdministrators: 'security_access_administrators',
+  accessGroups: 'security_access_groups',
+  accessGroupsRead: 'security_access_groups_read',
+  accessIndividuals: 'security_access_individuals',
+  accessIndividualsRead: 'security_access_individuals_read',
+  created: 'created',
+  filterTemplate: 'filter_template',
+  modified: 'modified',
+  owner: 'owner',
+  querySettings: 'query_settings',
+  id: 'id',
+  title: 'title',
+}
+
+const untransformAttrs = inverseKeys(transformAttrs)
+
+const renameKeys = (f, map) => {
+  return Object.keys(map).reduce((attrs, attr) => {
+    const name = f(attr)
+    attrs[name] = map[attr]
+    return attrs
+  }, {})
+}
+
+const queryTemplates = async (parent, args, { fetch }) => {
+  const res = await fetch(`${ROOT}/forms/query`)
+  const json = await res.json()
+  const queryTemplates = json
+    .map(attrs => renameKeys(k => transformAttrs[k], attrs))
+    .map(({ modified, created, ...rest }) => {
+      return {
+        ...rest,
+        created: new Date(created).toISOString(),
+        modified: new Date(modified).toISOString(),
+      }
+    })
+
+  //Any reason this is necessary?
+  // const status = {
+  //   // count: Int
+  //   // elapsed: Int
+  //   // hits: Int
+  //   // id: ID
+  //   // successful: Boolean
+  //   count: queryTemplates.length,
+  //   successful: true,
+  //   hits: queryTemplates.length,
+  // }
+  return queryTemplates
+}
+
+const saveQueryTemplate = async (parent, args, { fetch }) => {
+  const { queryTemplate } = args
+  const { id } = queryTemplate
+
+  const res = await fetch(`${ROOT}/forms/query/${id}`, {
```

Comment:
can we not use `catalog.create` here?

Reply:
I'm not really sure how all the [SearchFormsApplication](https://github.com/codice/ddf/blob/5980153da1c17d03b1e5672930fbf8a125e79e3f/catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/SearchFormsApplication.java) logic works. I'm going to try to convert to using the `catalog` methods over break